toothpick-runtime/src/main/java/toothpick/ScopeImpl.java

```diff
@@ -195,34 +195,34 @@ private void installModule(Module module) {
    * it will be added to the pool of un-scoped providers.
    * Note that
    *
-   * @param clazz the {@link Class} of {@code T} for which we lookup an {@link UnScopedProviderImpl}.
+   * @param clazz the {@link Class} of {@code T} for which we lookup an {@link InternalProviderImpl}.
    * @param name the potential name of the provider when it was bound (which means we always returned a scoped provider if
    * name is not null).
-   * @param <T> the type for which we lookup an {@link UnScopedProviderImpl}.
-   * @return a provider associated to the {@code T}. The returned provider is un-scoped (remember that {@link ScopedProviderImpl} is a subclass of
-   * {@link UnScopedProviderImpl}). The returned provider will be scoped by the public methods to use the current scope.
+   * @param <T> the type for which we lookup an {@link InternalProviderImpl}.
+   * @return a provider associated to the {@code T}. The returned provider is un-scoped.
+   * The returned provider will be scoped by the public methods to use the current scope.
    */
-  private <T> UnScopedProviderImpl<? extends T> getProviderInternal(Class<T> clazz, String name) {
+  private <T> InternalProviderImpl<? extends T> getProviderInternal(Class<T> clazz, String name) {
     if (clazz == null) {
       throw new IllegalArgumentException("TP can't get an instance of a null class.");
     }
     synchronized (clazz) {
-      ScopedProviderImpl<? extends T> scopedProvider = getScopedProvider(clazz, name);
+      InternalProviderImpl<? extends T> scopedProvider = getScopedProvider(clazz, name);
```

Comment:
I suggest we rename it to getBoundProvider. And in the old logic, it should have returned a UnScopedProvider, not a ScopedPovider. In the case we keep Scoped & UnScoped providers.